server/aap/commands/fulfill_image_assignments.py

```diff
@@ -183,28 +188,75 @@ def _get_image_modifier(self, assignment):
         # return the assignor user or the assignor desk, if nothing else available
         return {'proxy_user': assigned_to.get('assignor_user', assigned_to.get('assignor_desk'))}
 
+    def _extract_user(self, image):
+        """
+        Given an image dossier extract the user that modified it
+        :param image:
+        :return:
+        """
+        dossier = image.find('./dc_rest_docs/dc_rest_doc/dcdossier')
+        if dossier is not None:
+            modified_by = dossier.attrib.get('modified_by')
+        if modified_by is not None and not modified_by == 'system':
+            user_service = superdesk.get_resource_service('users')
+            user = user_service.find_one(req=None, username=modified_by)
+            if user:
+                return user
+        return None
+
     def _check_in_progress(self, assignments):
         """
         Check the AAP Image Pool for any assignments that may be in progress
         :param assignments:
         :return:
         """
         in_progress_assignments = list()
+        in_list = set()
         for assignment in assignments:
-            # look in the AAP Image pool
-            dc_images = self._get_dc_images_by_field('aapimage', assignment.get('_id'))
-            if dc_images:
-                count = dc_images.find('./doc_count')
-                if count and int(count.text) > 0:
-                    in_progress_assignments.append(assignment)
+            if assignment.get('assigned_to', {}).get('state') in ASSIGNMENT_WORKFLOW_STATE.ASSIGNED:
+                # look in the AAP Image pool
+                dc_images = self._get_dc_images_by_field('aapimage', assignment.get('_id'))
+                if dc_images is not None:
+                    count = dc_images.find('./doc_count')
+                    if count is not None and int(count.text) > 0:
+                        logger.info('Found in progress items in aapimage for {}'.format(assignment.get('_id')))
+                        user = self._extract_user(dc_images)
+                        in_list.add(assignment.get('_id'))
+                        in_progress_assignments.append({'assignment': assignment, 'user': user})
```

Comment:
If images are found in `aapimage`, should a `continue` statement be used here so `picedit` is not queried?

Reply:
could do!